src/services/notificationService.ts

```diff
@@ -10,10 +11,37 @@ import { logger } from '../utils/logger';
 import { EMAIL_STATUSES, Notification } from '../entities/notification';
 import { SEGMENT_METADATA_SCHEMA_VALIDATOR } from '../utils/validators/segmentAndMetadataValidators';
 import { validateWithJoiSchema } from '../validators/schemaValidators';
-import { SegmentAnalyticsSingleton } from './segment/segmentAnalyticsSingleton';
 import { SendNotificationRequest } from '../types/requestResponses';
 import { StandardError } from '../types/StandardError';
 
+enum EmailNotificationId {
+  donationReceived = 'donationReceived',
+}
+
+const activityCreator = (payload: any, emailNotificationId: EmailNotificationId) => {
+  if (emailNotificationId === EmailNotificationId.donationReceived) {
+    return {
+      "activities": [
+        {
+          "activity_id": `act:cm:${emailNotificationId}`,
+          "attributes": {
+            "str:cm:projecttitle": payload.title,
+            "int:cm:donationamount": payload.amount,
+            "str:cm:donationtoken": payload.token,
+            "str:cm:email": payload.email,
+            "str:cm:projectlink": payload.projectLink,
+            "bol:cm:verified": payload.verified,
+            "str:cm:transactionlink": payload.transactionLink,
+          },
+          "fields": {
+            "str::email": payload.email
+          }
+        }
+      ]
+    };
+  }
+}
+
 export const sendNotification = async (
```

Comment:
Please create emailAdapter and move this function to that file, we want to make this part agnostic about our email service API call